driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -404,6 +397,7 @@ void cancel() {
 
         @Override
         public Message.Request request() {
+            Message.Request request = callback.request();
```

Comment:
Is the idea to always copy because the request could be used again in a Retry (independent of a speculative execution being executed?)